zappnode-front/src/routes/Register.js

```diff
@@ -15,12 +15,12 @@ function Register() {
   const [passwordWarning, setPasswordWarning] = useState(<ul></ul>)
 
   const validEmailAdress = () => {
-    const re = /^(([^<>()\[\]\\.,;:\s@"]+(\.[^<>()\[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/
+    const re = /[^@ \t\r\n]+@[^@ \t\r\n]+\.[^@ \t\r\n]+/
     return re.test(email);
   };
 
   const validPassword = () => {
-    const re = /^([a-zA-Z0-9!@#$%^&*]{8,15})$/;
+    const re = /^(?=.*?[A-Z])(?=.*?[a-z])(?=.*?[0-9])(?=.*?[#?!@$ %^&*-]).{8,}$/;
```

Comment:
I think we should have a max length for passwords to prevent insanely long passwords that take up space and data unnecessarily.